### PR TITLE
Redirect output to /dev/null

### LINKED ISSUE #2877: Textual interferes with other libraries' use of logging
I'm trying to create a Textual app that creates an instance of an object that eventually loads the [`transformers` module](https://pypi.org/project/transformers/), and this fails due to something that module is doing with the logging module. Here's a minimal example that demonstrates the issue:

```python
import logging, sys
from textual.app import App
from textual.logging import TextualHandler

logging.basicConfig(
    level="NOTSET",
    handlers=[TextualHandler()],
)

class OtherClass:
    def __init__(self):
        # https://github.com/huggingface/transformers/blob/main/src/transformers/utils/logging.py#L87
        _default_handler = logging.StreamHandler()  # Set sys.stderr as stream.                                                        │
        _default_handler.flush = sys.stderr.flush

class MyApp(App):
    def on_mount(self) -> None:
        self.other = OtherClass()

if __name__ == "__main__":
    app = MyApp()
    app.run()
```

This fails with the error `AttributeError: 'NoneType' object has no attribute 'flush'`.

`TextualHandler` doesn't alter this behavior. The workaround I'm currently using is defining the instance of OtherClass in the main, before creating MyApp, but that does make some of the code a lot more cumbersome and it shouldn't be necessary.

### PR DESCRIPTION
**Pull request by alexiri**
Fixes #2877, #2878 and #2879.

**Please review the following checklist.**

- [ ] Docstrings on all new or modified functions / classes 
- [ ] Updated documentation
- [ ] Updated CHANGELOG.md (where appropriate)

### CODE REVIEW

File: src/textual/app.py [hunk 1]
```diff
@@ -1978,8 +1978,8 @@ async def invoke_ready_callback() -> None:
                                 with redirect_stdout(redirector):  # type: ignore
                                     await run_process_messages()
                         else:
-                            with redirect_stderr(None):
-                                with redirect_stdout(None):
+                            with redirect_stderr(open(os.devnull, 'w')):
```
Comment (willmcgugan) [inline src/textual/app.py:1981]: The file handle should really be explicitly closed. You could probably get away with opening a single file.

Comment (alexiri) [reply src/textual/app.py]: You're absolutely right, sorry about that.

Comment (willmcgugan) [reply src/textual/app.py]: No worries!

File: src/textual/app.py [hunk 2]
```diff
@@ -1978,8 +1978,8 @@ async def invoke_ready_callback() -> None:
                                 with redirect_stdout(redirector):  # type: ignore
                                     await run_process_messages()
                         else:
-                            with redirect_stderr(None):
-                                with redirect_stdout(None):
+                            with open(os.devnull, "w") as thevoid:
```
Comment (willmcgugan) [inline src/textual/app.py:1981]: We prefer camel case words. I think there is a more apt term than "void" here. Suggest `null_file`.

**Review by willmcgugan — CHANGES_REQUESTED** [review]

File: src/textual/app.py (hunk 2 above)
Comment (alexiri) [reply src/textual/app.py]: `null_file` makes the line too long and `black` complains. Should I reflow the line even though it's ugly, or should I use the "standard" `f` as a file handle?

Comment (willmcgugan) [reply src/textual/app.py]: I'd break it out on to another line. Don't mind that. No single letter variables though...

Comment (alexiri) [reply src/textual/app.py]: Ok, done.

**Review by willmcgugan — APPROVED** [review]
Thanks

**Pull request merged**